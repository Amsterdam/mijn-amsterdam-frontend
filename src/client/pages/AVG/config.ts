import { AVGRequestFrontend } from '../../../server/services/avg/types';
import { dateSort } from '../../../universal/helpers/date';
import { LinkProps } from '../../../universal/types';
import {
  DisplayProps,
  WithDetailLinkComponent,
} from '../../components/Table/TableV2';

export const displayPropsAanvragen: DisplayProps<
  WithDetailLinkComponent<AVGRequestFrontend>
> = {
  detailLinkComponent: 'Nummer',
  ontvangstDatumFormatted: 'Ontvangen op',
  themas: 'Onderwerp(en)',
};

export const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
};

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
};

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    title: 'Lopende aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => !avgVerzoek.datumAfhandeling,
    ...tableConfigBase,
  },
  [listPageParamKind.completed]: {
    title: 'Afgehandelde aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => avgVerzoek.datumAfhandeling,
    ...tableConfigBase,
  },
};

export const linkListItems: LinkProps[] = [
  {
    to: 'https://www.amsterdam.nl/privacy/loket/',
    title: 'Loket persoonsgegevens gemeente Amsterdam',
  },
];
