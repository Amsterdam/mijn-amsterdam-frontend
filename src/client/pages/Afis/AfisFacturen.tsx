import { useParams } from 'react-router-dom';

import { AfisDisclaimer, AfisDisclaimerOvergedragenFacturen } from './Afis';
import styles from './Afis.module.scss';
import { useAfisListPageData } from './useAfisThemaData.hook';
import { AfisFactuurState } from '../../../server/services/afis/afis-types';
import { ListPagePaginated } from '../../components/ListPagePaginated/ListPagePaginated';
import { PageContentCell } from '../../components/Page/Page';

function AfisListPageBody({ state }: { state: AfisFactuurState }) {
  switch (state) {
    case 'open':
      return <AfisDisclaimer />;
    case 'overgedragen':
      return <AfisDisclaimerOvergedragenFacturen />;
    default:
      return null;
  }
}

export function AfisFacturen() {
  const { state } = useParams<{ state: AfisFactuurState }>();
  const {
    facturenListResponse,
    isListPageError,
    isListPageLoading,
    facturenTableConfig,
    isThemaPaginaError,
    isThemaPaginaLoading,
    routes,
    themaPaginaBreadcrumb,
  } = useAfisListPageData(state);

  const listPageTableConfig = facturenTableConfig[state];
  const facturen = facturenListResponse?.facturen ?? [];

  return (
    <ListPagePaginated
      items={facturen}
      pageContentTop={
        <PageContentCell>
          <AfisListPageBody state={state} />
        </PageContentCell>
      }
      title={listPageTableConfig.title}
      appRoute={routes.listPageFacturen}
      appRouteParams={{ state }}
      breadcrumbs={[themaPaginaBreadcrumb]}
      displayProps={listPageTableConfig.displayProps}
      isLoading={isThemaPaginaLoading || isListPageLoading}
      isError={isThemaPaginaError || isListPageError}
      tableClassName={styles[listPageTableConfig.className]}
    />
  );
}
