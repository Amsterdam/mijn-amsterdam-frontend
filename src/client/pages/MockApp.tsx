import { RecoilRoot, RecoilRootProps } from 'recoil';
import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';

interface MockAppProps {
  routePath: string;
  routeEntry: string;
  initializeState?: RecoilRootProps['initializeState'];
  component: React.Component | any;
}

export default function MockApp({
  routePath,
  routeEntry,
  initializeState,
  component,
}: MockAppProps) {
  return (
    <RecoilRoot initializeState={initializeState}>
      <MemoryRouter initialEntries={[routeEntry]}>
        <Route path={routePath} component={component} />
      </MemoryRouter>
    </RecoilRoot>
  );
}
