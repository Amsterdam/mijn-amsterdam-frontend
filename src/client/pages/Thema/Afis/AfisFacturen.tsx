import { useParams } from 'react-router';

import { AfisDisclaimer, AfisDisclaimerOvergedragenFacturen } from './Afis';
import { useAfisListPageData } from './useAfisThemaData.hook';
import { AfisFactuurState } from '../../../../server/services/afis/afis-types';
import { ListPagePaginated } from '../../../components/ListPagePaginated/ListPagePaginated';
import { PageContentCell } from '../../../components/Page/Page';

function AfisListPageBody({ state }: { state: AfisFactuurState }) {
  switch (state) {
    case 'open':
      return <AfisDisclaimer />;
    case 'overgedragen':
      return <AfisDisclaimerOvergedragenFacturen />;
    default:
      return null;
  }
}

export function AfisFacturen() {
  const { state = 'open' } = useParams<{ state: AfisFactuurState }>();
  const {
    facturenListResponse,
    isListPageError,
    isListPageLoading,
    facturenTableConfig,
    isThemaPaginaError,
    isThemaPaginaLoading,
    routeConfig,
    breadcrumbs,
  } = useAfisListPageData(state);

  const listPageTableConfig = facturenTableConfig[state];
  const facturen = facturenListResponse?.facturen ?? [];

  return (
    <ListPagePaginated
      items={facturen}
      pageContentTop={
        <PageContentCell>
          <AfisListPageBody state={state} />
        </PageContentCell>
      }
      title={listPageTableConfig.title}
      appRoute={routeConfig.listPage.path}
      appRouteParams={{ state }}
      breadcrumbs={breadcrumbs}
      displayProps={listPageTableConfig.displayProps}
      isLoading={isThemaPaginaLoading || isListPageLoading}
      isError={isThemaPaginaError || isListPageError}
    />
  );
}
