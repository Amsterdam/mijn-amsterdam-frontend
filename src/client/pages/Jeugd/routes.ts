import { JeugdDetail } from './JeugdDetail';
import { JeugdThemaPagina } from './JeugdThema';
import { FeatureToggle } from '../../../universal/config/feature-toggles';

const baseRoute = '/jeugd';

export const JeugdRoutes = [
  {
    route: baseRoute,
    Component: JeugdThemaPagina,
    isActive: FeatureToggle.zorgnedLeerlingenvervoerActive,
  },
  {
    route: `${baseRoute}/voorziening/:id`,
    Component: JeugdDetail,
    isActive: FeatureToggle.zorgnedLeerlingenvervoerActive,
  },
];
