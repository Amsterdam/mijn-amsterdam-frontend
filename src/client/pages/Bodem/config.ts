import { generatePath } from 'react-router-dom';

import { LoodMetingFrontend } from '../../../server/services/bodem/types';
import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { LinkProps } from '../../../universal/types';

const displayPropsAanvragen = {
  detailLinkComponent: 'Adres',
  datumAanvraagFormatted: 'Aangevraagd',
  status: 'Status',
};

const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
};

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
};

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    ...tableConfigBase,
    title: 'Lopende aanvragen',
    filter: (bodemAanvraag: LoodMetingFrontend) => !bodemAanvraag.processed,
    listPageRoute: generatePath(AppRoutes['BODEM/LIST'], {
      kind: listPageParamKind.inProgress,
    }),
  },
  [listPageParamKind.completed]: {
    ...tableConfigBase,
    title: 'Afgehandelde aanvragen',
    filter: (bodemAanvraag: LoodMetingFrontend) => bodemAanvraag.processed,
    listPageRoute: generatePath(AppRoutes['BODEM/LIST'], {
      kind: listPageParamKind.completed,
    }),
  },
} as const;

export const linkListItems: LinkProps[] = [
  {
    title: 'Meer informatie over lood in de bodem.',
    to: 'https://www.amsterdam.nl/wonen-leefomgeving/bodem/lood-grond/',
  },
] as const;
