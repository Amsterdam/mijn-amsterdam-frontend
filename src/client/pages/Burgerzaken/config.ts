import { generatePath } from 'react-router-dom';

import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { IdentiteitsbewijsFrontend, LinkProps } from '../../../universal/types';
import { withOmitDisplayPropsForSmallScreens } from '../../components/Table/helpers';
import {
  DisplayProps,
  WithDetailLinkComponent,
} from '../../components/Table/TableV2';

const displayPropsBase: DisplayProps<
  WithDetailLinkComponent<IdentiteitsbewijsFrontend>
> = {
  detailLinkComponent: 'Type',
  datumAfloopFormatted: 'Geldig tot',
};

const listPageParamKind = {
  identiteitsbewijzen: 'identiteitsbewijzen',
};

const displayProps = withOmitDisplayPropsForSmallScreens(displayPropsBase, [
  'datumAfloopFormatted',
]);

export const tableConfig = {
  [listPageParamKind.identiteitsbewijzen]: {
    title: 'Mijn reisdocumenten',
    displayProps,
    sort: dateSort('datumAfloop', 'desc'),
    listPageRoute: generatePath(AppRoutes['BURGERZAKEN/LIST'], {
      kind: listPageParamKind.identiteitsbewijzen,
    }),
  },
} as const;

export const linkListItems: LinkProps[] = [
  {
    to: 'https://www.amsterdam.nl/burgerzaken',
    title: 'Overzicht en aanvragen bij burgerzaken',
  },
] as const;

export const backLinkDetailPage = AppRoutes.BURGERZAKEN;
