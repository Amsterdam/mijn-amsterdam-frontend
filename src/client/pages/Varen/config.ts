import { generatePath } from 'react-router-dom';

import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { VarenFrontend } from '../../../server/services/varen/config-and-types';

export const displayPropsAanvragen = {
  detailLinkComponent: '',
  title: 'Omschrijving',
  dateRequestFormatted: 'Aangevraagd',
  status: 'Status',
};

export const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
};

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
};

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    title: 'Lopende aanvragen',
    filter: (vergunning: VarenFrontend) => !vergunning.dateEnd,
    listPageRoute: generatePath(AppRoutes['VAREN/LIST'], {
      kind: listPageParamKind.inProgress,
    }),
    ...tableConfigBase,
  },
  [listPageParamKind.completed]: {
    title: 'Afgehandelde aanvragen',
    filter: (vergunning: VarenFrontend) => vergunning.dateEnd,
    listPageRoute: generatePath(AppRoutes['VAREN/LIST'], {
      kind: listPageParamKind.completed,
    }),
    ...tableConfigBase,
  },
};
