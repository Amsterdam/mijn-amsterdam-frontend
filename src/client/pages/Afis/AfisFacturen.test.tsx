import { render, waitFor } from '@testing-library/react';
import { generatePath } from 'react-router-dom';
import { MutableSnapshot } from 'recoil';

import { bffApi } from '../../../test-utils';
import { AppRoutes } from '../../../universal/config/routes';
import { AppState } from '../../../universal/types';
import { appStateAtom } from '../../hooks/useAppState';
import MockApp from '../MockApp';
import { AfisFacturen } from './AfisFacturen';

const businessPartnerIdEncrypted = 'yyy-456-yyy';
const testState = {
  AFIS: {
    status: 'OK',
    content: {
      isKnown: true,
      businessPartnerIdEncrypted,
    },
  },
} as AppState;

function initializeState(snapshot: MutableSnapshot) {
  snapshot.set(appStateAtom, testState);
}

describe('<AfisFacturen />', () => {
  const routePath = AppRoutes['AFIS/FACTUREN'];

  test('Lists Open facturen', async () => {
    bffApi
      .get(`/services/afis/facturen/overzicht?id=${businessPartnerIdEncrypted}`)
      .reply(200, {
        content: {
          open: {
            count: 1,
            facturen: [
              {
                id: '2',
                factuurDocumentId: '2',
                afzender: 'Company E',
                datePublished: '2023-04-15',
                datePublishedFormatted: '15 april 2023',
                paymentDueDate: '2023-05-15',
                paymentDueDateFormatted: '15 mei 2023',
                debtClearingDate: null,
                debtClearingDateFormatted: null,
                amountOwed: 2500,
                amountOwedFormatted: '€ 2.500,00',
                factuurNummer: 'F005',
                factuurNummerEl: 'F005',
                status: 'openstaand',
                statusDescription: 'Openstaand description',
                paylink: 'https://payment.example.com/F005',
                documentDownloadLink: 'https://download.example.com/F005',
              },
            ],
          },
          afgehandeld: { facturen: [], count: 0 },
          overgedragen: { facturen: [], count: 0 },
        },
        status: 'OK',
      });

    const routeEntry = generatePath(routePath, {
      state: 'open',
    });

    const Component = () => (
      <MockApp
        routeEntry={routeEntry}
        routePath={routePath}
        component={AfisFacturen}
        initializeState={initializeState}
      />
    );
    const screen = render(<Component />);

    await waitFor(() => {
      expect(screen.getByText('15 mei 2023')).toBeInTheDocument();
      expect(screen.getByText('Openstaand description')).toBeInTheDocument();
      expect(screen.getByText('Company E')).toBeInTheDocument();
    });
  });

  test('Lists Closed facturen', async () => {
    bffApi
      .get(
        `/services/afis/facturen/afgehandeld?id=${businessPartnerIdEncrypted}`
      )
      .reply(200, {
        content: {
          afgehandeld: {
            count: 1,
            facturen: [
              {
                id: '1',
                factuurDocumentId: '1',
                afzender: 'Company D',
                datePublished: '2023-04-01',
                datePublishedFormatted: '1 april 2023',
                paymentDueDate: '2023-05-01',
                paymentDueDateFormatted: '1 mei 2023',
                debtClearingDate: null,
                debtClearingDateFormatted: null,
                amountOwed: 1500,
                amountOwedFormatted: '€ 1.500,00',
                factuurNummer: 'F001',
                factuurNummerEl: 'F001',
                status: 'betaald',
                statusDescription: 'Betaalde description',
                paylink: null,
                documentDownloadLink: 'https://download.example.com/F004',
              },
            ],
          },
        },
        status: 'OK',
      });

    bffApi
      .get(`/services/afis/facturen/overzicht?id=${businessPartnerIdEncrypted}`)
      .reply(200, {
        content: {
          open: { count: 0, facturen: [] },
          afgehandeld: { facturen: [], count: 0 },
          overgedragen: { facturen: [], count: 0 },
        },
        status: 'OK',
      });

    const routeEntry = generatePath(routePath, {
      state: 'afgehandeld',
    });

    const Component = () => (
      <MockApp
        routeEntry={routeEntry}
        routePath={routePath}
        component={AfisFacturen}
        initializeState={initializeState}
      />
    );
    const screen = render(<Component />);

    await waitFor(() => {
      expect(screen.getByText('Company D')).toBeInTheDocument();
      expect(screen.getByText('Betaalde description')).toBeInTheDocument();
      expect(screen.queryByText('15 mei 2023')).not.toBeInTheDocument();
    });
  });

  test('Partial error display', () => {});

  test('Error display', () => {});
});
