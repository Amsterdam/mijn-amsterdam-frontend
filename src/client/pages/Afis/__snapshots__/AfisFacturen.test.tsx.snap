// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AfisFacturen /> > Matches the Full Page snapshot 1`] = `
<DocumentFragment>
  <main
    class="_Page_47d7c0 _OverviewPage_47d7c0"
  >
    <header
      class="_PageHeading_3b337d _hasIcon_3b337d _hasBackLink_3b337d"
    >
      <span
        aria-hidden="true"
        class="_Icon_3b337d"
      >
        <svg
          aria-label="Facturen en betalen"
          class="_ThemaIcon_2fc54d"
          fill="black"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0_17050_1622)"
          >
            <path
              d="M4 24C2.93333 24 2 23.64 1.2 22.92C0.4 22.2 0 21.36 0 20.4V16.8H4V0L6 1.8L8 0L10 1.8L12 0L14 1.8L16 0L18 1.8L20 0L22 1.8L24 0V20.4C24 21.36 23.6 22.2 22.8 22.92C22 23.64 21.0667 24 20 24H4ZM20 21.6C20.4 21.6 20.6667 21.48 20.9333 21.24C21.2 21 21.3333 20.76 21.3333 20.4V3.6H6.66667V16.8H18.6667V20.4C18.6667 20.76 18.8 21 19.0667 21.24C19.3333 21.48 19.6 21.6 20 21.6ZM8 8.4V6H16V8.4H8ZM8 12V9.6H16V12H8ZM18.6667 8.4C18.2667 8.4 18 8.28 17.7333 8.04C17.4667 7.8 17.3333 7.56 17.3333 7.2C17.3333 6.84 17.4667 6.6 17.7333 6.36C18 6.12 18.2667 6 18.6667 6C19.0667 6 19.3333 6.12 19.6 6.36C19.8667 6.6 20 6.84 20 7.2C20 7.56 19.8667 7.8 19.6 8.04C19.3333 8.28 19.0667 8.4 18.6667 8.4ZM18.6667 12C18.2667 12 18 11.88 17.7333 11.64C17.4667 11.4 17.3333 11.16 17.3333 10.8C17.3333 10.44 17.4667 10.2 17.7333 9.96C18 9.72 18.2667 9.6 18.6667 9.6C19.0667 9.6 19.3333 9.72 19.6 9.96C19.8667 10.2 20 10.44 20 10.8C20 11.16 19.8667 11.4 19.6 11.64C19.3333 11.88 19.0667 12 18.6667 12ZM4 21.6H16V19.2H2.66667V20.4C2.66667 20.76 2.8 21 3.06667 21.24C3.33333 21.48 3.6 21.6 4 21.6Z"
            />
          </g>
          <defs>
            <clippath
              id="clip0_17050_1622"
            >
              <rect
                fill="white"
                height="24"
                width="24"
              />
            </clippath>
          </defs>
        </svg>
      </span>
      <div
        class="_HeadingInner_3b337d"
      >
        <a
          class="_Button_c3ab45 _Button__plain_c3ab45 _Button__lean_c3ab45 _Linkd_c3ab45 _BackLink_3b337d"
          href="/afis"
        >
          <svg
            aria-hidden="true"
            aria-label="chevron-links-icoon"
            class="_Icon_c3ab45 _Icon__left_c3ab45"
            fill="#000"
            focusable="false"
            height="14px"
            viewBox="0 0 32 32"
            width="14px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <polygon
              points="22.8571429 32 6.85714286 16 22.8571429 0 25.7568858 2.91002571 12.6771943 16 25.7568858 29.0899743"
            />
          </svg>
          Openstaande facturen
        </a>
        <h2
          class="ams-heading ams-heading--1"
        >
          Openstaande facturen
        </h2>
      </div>
    </header>
    <div
      class="ams-screen ams-screen--wide"
    >
      <div
        class="ams-grid"
      >
        <div
          class="ams-grid__cell ams-grid__cell--span-all"
        >
          <span
            class="_LoadingContent_16e37c"
          >
            <span
              style="width: 100%; height: 2rem; margin-bottom: 2rem;"
            />
            <span
              style="width: 100%; height: 2rem; margin-bottom: 2rem;"
            />
            <span
              style="width: 100%; height: 2rem; margin-bottom: 2rem;"
            />
          </span>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
